### PR TITLE
e2e fixtures and WS

### PR DESCRIPTION
**Pull request by dsesami**
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION

```
awx: 3.0.1
```

### CODE REVIEW

File: awx/ui/test/e2e/commands/findThenClick.js [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
-
 /* Utility function for clicking elements; attempts to scroll to
  * the element if necessary, and waits for the page to finish loading.
  *
- * @param selector - xpath of the element to click. */
-exports.command = function findThenClick (selector) {
+ * @param selector - xpath or css selector of the element to click.
+ * @param [locatoryStrategy='xpath'] - locator strategy used. */
+
+exports.command = function findThenClick (selector, locatorStrategy = 'xpath') {
```
Comment (elyezer) [inline awx/ui/test/e2e/commands/findThenClick.js:7]: should this change the locator strategy inside this function and revert it before returning?

File: awx/ui/test/e2e/commands/navigateTo.js [hunk 2]
```diff
@@ -5,8 +5,9 @@ exports.command = function navigateTo (url, expectSpinny = true) {
     this.url(url);
 
     if (expectSpinny) {
-        this.waitForElementVisible(spinny);
-        this.waitForElementNotVisible(spinny);
+        this.waitForElementVisible(spinny, () => {
```
Comment (elyezer) [inline awx/ui/test/e2e/commands/navigateTo.js:8]: maybe `waitForSpinny` could be used here? https://github.com/ansible/awx/blob/devel/awx/ui/test/e2e/commands/waitForSpinny.js#L2

File: awx/ui/test/e2e/tests/test-websockets.js [hunk 3]
```diff
@@ -48,28 +53,101 @@ module.exports = {
...
+            .navigateTo(`${AWX_E2E_URL}/#/home`);
         getJob('test-websockets', 'debug.yml', 'test-websockets-successful');
-        client.expect.element(spinny).to.not.be.visible.before(5000);
-        client.expect.element(`${sparklineIcon}[1]${running}`)
-            .to.be.visible.before(5000);
 
-        // Allow a maximum amount of 30 seconds for the job to complete.
-        client.expect.element(`${successfulJt}${sparklineIcon}[1]${success}`)
-            .to.be.present.after(30000);
+        client.expect.element('.spinny')
```
Comment (elyezer) [inline awx/ui/test/e2e/tests/test-websockets.js:61]: Could `waitForSpinny` be used here?

**Review by elyezer — APPROVED** [review]
LGTM, left some non-blocking comments

File: awx/ui/test/e2e/commands/findThenClick.js (hunk 1 above)
Comment (dsesami) [reply awx/ui/test/e2e/commands/findThenClick.js:7]: Despite several attempts to hunt for a solution, there isn't a pleasant way to detect the current locator strategy, so I opted for a way to pass in what strategy was being used manually.

File: awx/ui/test/e2e/commands/navigateTo.js (hunk 2 above)
Comment (dsesami) [reply awx/ui/test/e2e/commands/navigateTo.js:8]: that function isn't locator-strategy agnostic, I'll see if I can refactor it to default to its usual use, and offer a param here like I did elsewhere.

File: awx/ui/test/e2e/commands/findThenClick.js [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
     this.waitForElementPresent(selector, () => {
         this.moveToElement(selector, 0, 0, () => {
             this.click(selector, () => {
-                this.waitForElementNotVisible(spinny);
+                if (locatorStrategy === 'css') {
```
Comment (unlikelyzero) [inline awx/ui/test/e2e/commands/findThenClick.js:11]: awesome

File: awx/ui/test/e2e/fixtures.js [hunk 5]
```diff
@@ -288,16 +288,18 @@ const waitForJob = endpoint => {
...
+        getProject(namespace),
+    ];
+    return Promise.all(promises)
+        .then(([project]) => {
+            post(`/api/v2/projects/${project.id}/update/`, {})
+                .then(update => waitForJob(update.data.url))
+                .then(() => { project = getProject(namespace); });
+            return project;
+        });
+};
```
Comment (unlikelyzero) [inline awx/ui/test/e2e/fixtures.js:302]: Because you're modifying a fixture, you should make sure that the rest of the regression suite passes

**Review by unlikelyzero — APPROVED** [review]
🎉

**Pull request merged**